src/test/java/htsjdk/samtools/util/BlockCompressedInputStreamTest.java

```diff
@@ -86,4 +91,142 @@ public void available_should_return_number_of_bytes_left_in_current_block() thro
+        final StringBuilder sb = new StringBuilder(BlockCompressedStreamConstants.DEFAULT_UNCOMPRESSED_BLOCK_SIZE * 2);
+        s = "1234567890123456789012345678901234567890123456789012345678901234567890\n";
+        while (sb.length() <= BlockCompressedStreamConstants.DEFAULT_UNCOMPRESSED_BLOCK_SIZE) {
+            sb.append(s);
+            linesWritten.add(s);
+        }
+        bcos.write(sb.toString().getBytes()); //Call 3
+        bcos.close();
+
+        int[] inflateCalls = {0}; //Note: using and array is a HACK to fool the compiler
+        class MyInflater extends Inflater {
+            MyInflater(boolean gzipCompatible){
+                super(gzipCompatible);
+            }
+            @Override
+            public int inflate(byte[] b, int off, int len) throws java.util.zip.DataFormatException {
+                inflateCalls[0]++;
+                return super.inflate(b, off, len);
+            }
+        }
+        final InflaterFactory myInflaterFactory = new InflaterFactory() {
+            public Inflater makeInflater(final boolean gzipCompatible) {
+                return new MyInflater(gzipCompatible);
+            }
+        };
+
+        BlockGunzipper.setDefaultInflaterFactory(myInflaterFactory);
+        BlockCompressedInputStream bcis = new BlockCompressedInputStream(new FileInputStream(f), false);
+        BufferedReader reader = new BufferedReader(new InputStreamReader(bcis));
+        String line;
+        for(int i = 0; (line = reader.readLine()) != null; ++i) {
+            Assert.assertEquals(line + "\n", linesWritten.get(i));
+        }
+        bcis.close();
+        Assert.assertEquals(inflateCalls[0], 4, "inflate calls");
+    }
+
+    @Test
+    public void testCustomInflaterUrlInput() throws Exception {
+        int[] inflateCalls = {0}; //Note: using and array is a HACK to fool the compiler
+        class MyInflater extends Inflater {
+            MyInflater(boolean gzipCompatible){
+                super(gzipCompatible);
+            }
+            @Override
+            public int inflate(byte[] b, int off, int len) throws java.util.zip.DataFormatException {
+                inflateCalls[0]++;
+                return super.inflate(b, off, len);
+            }
+        }
+        final InflaterFactory myInflaterFactory = new InflaterFactory() {
+            public Inflater makeInflater(final boolean gzipCompatible) {
+                return new MyInflater(gzipCompatible);
+            }
+        };
+
+        BlockGunzipper.setDefaultInflaterFactory(myInflaterFactory);
+        BlockCompressedInputStream bcis = new BlockCompressedInputStream(new URL("http://broadinstitute.github.io/picard/testdata/index_test.bam"));
+        BufferedReader reader = new BufferedReader(new InputStreamReader(bcis));
+        String line;
+        for(int i = 0; (line = reader.readLine()) != null; ++i) {}
+        bcis.close();
+        Assert.assertEquals(inflateCalls[0], 21, "inflate calls");
+    }
 }
```

Comment:
Can you add a test case that covers an `AsyncBlockCompressedInputStream` with a custom inflater as well?